crates/admin/src/cluster_controller/logs_controller/nodeset_selection.rs

```diff
@@ -141,11 +185,27 @@ impl<'a> NodeSetSelector<'a> {
     }
 }
 
+#[derive(Debug)]
+struct NodeSetSizeRange {
+    /// Absolute minimum size of the nodeset
+    /// That satisfies the replication property
+    /// with given strategy.
+    ///
+    /// Going below this minimum size will lose
+    /// write availability
+    degraded_size: usize,
+    /// The minimum number of nodes to satisfy
+    /// the replication property with given strategy
+    minimum_size: usize,
```

Comment:
I think because degraded is basically == "number of copies" and hence it's the minimum possible number of nodes to have so you still have write availability but you can't afford losing any more nodes.

I will rephrase it